components/ws-manager-bridge/src/bridge.ts

```diff
@@ -504,23 +504,37 @@ export class WorkspaceManagerBridge implements Disposable {
             const actuallyRunningInstances = await client.getWorkspaces(ctx, new GetWorkspacesRequest());
             actuallyRunningInstances.getStatusList().forEach((s) => runningInstancesIdx.delete(s.getId()));
 
+            // runningInstancesIdx only contains instances that ws-manager is not aware of
             for (const [instanceId, ri] of runningInstancesIdx.entries()) {
                 const instance = ri.latestInstance;
-                if (instance.status.phase !== "running") {
-                    log.debug({ instanceId }, "Skipping instance", {
-                        phase: instance.status.phase,
-                        creationTime: instance.creationTime,
-                        region: instance.region,
-                    });
-                    continue;
+                const phase = instance.status.phase;
+
+                // When ws-manager is not aware of the following instances,
+                // they should be marked as stopped.
+                // If 'stopping', we give 10 sec estimated grace period in case of a race condition
+                if (
+                    phase === "pending" ||
+                    phase === "creating" ||
+                    phase === "initializing" ||
+                    phase === "running" ||
+                    (phase === "stopping" &&
+                        instance.stoppingTime &&
+                        durationLongerThanSeconds(Date.parse(instance.stoppingTime), 10))
+                ) {
+                    log.info(
+                        { instanceId, workspaceId: instance.workspaceId },
+                        "Database says the instance is running, but wsman does not know about it. Marking as stopped in database.",
+                        { installation },
+                    );
+                    await this.markWorkspaceInstanceAsStopped(ctx, ri, new Date());
```

Comment:
What about adding a `continue` here? That would avoid the additional info "skipping instance" below.

Reply:
Thanks for the catch! And I should've removed the continue in the line below.